### ISSUE TITLE
Error: this API has been removed.

### ISSUE THREAD

**Issue by galipmedia** [dudik/twitch-m3u8#1]
I am getting this error as of today. I guess Twitch has removed that API :( Any chance of making this work with the "New Twitch API" ?

**Comment by dkonopka** [comment]
I'm requesting for updating to new V5 Twitch API too 🙏

**Comment by dudik** [comment]
I'm incredibly sorry that I didn't notice this issue earlier. I don't use this package anymore and I didn't get any notification about it.

Updating to the new Twitch API isn't possible as this API endpoint isn't officially supported.

You have probably already seen this issue: https://github.com/nicholastay/node-twitch-get-stream/issues/9
So you should probably use the youtube-dl Client-ID: `kimne78kx3ncx6brgo4mv6wki5h1ko`

The API endpoint also seems to work when using the oauth token instead of a Client-ID. I could potentially add this functionality, but I think that using the mentioned Client-ID is a viable option for this package. Tell me what you think.

**Comment by Coolred** [comment]
It works after using above Client-ID.

**Comment by Coolred** [comment]
Thanks for the information. It works after using the above Client-ID.

**Comment by M0opex** [comment]
Hey @Coolred, I'm glad it worked for you, but I have no idea how to run this JS, can you please give a quick tutorial. thanks

**Comment by Coolred** [comment]
Hi @M0opex, I didn't use this module before.

But after my testing, what you may encounter is the Streamer part. You need to modify the source code and add file extension '.m3u8' after the channel id.
For example: `'api/channel/hls'}/${id}`.m3u8

Hope this will solve your issue.

**Comment by M0opex** [comment]
@Coolred thanks for the reply boss,

i'm not very familiar with this kind of work, However, I tried but it didn't work, I think I'm missing something,

Can you show me an example of this channel, please https://www.twitch.tv/esl_csgo

I hope I can find the final answer here :)

**Comment by dudik** [comment]
@M0opex I merged the fix. Does it work now?

Thank you Coolred and Matthew.

**Comment by Coolred** [comment]
@dudik It's working perfectly, but I think you did not release it yet.

@M0opex You can try the following sample code by overriding the module with latest index.js
```
// replace clientID with your Twitch Client ID
const twitch = require('twitch-m3u8')('kimne78kx3ncx6brgo4mv6wki5h1ko');

// returns a JSON object containing available streams of a livestream
twitch.getStream('esl_csgo')
.then(data => console.log(data))
.catch(err => console.error(err));
```

**Comment by dudik** [comment]
It should be published now. Looks like I forgot how to use npm.. :smiley: Thanks again.

**Comment by Coolred** [comment]
@dudik I saw the fix on 1.0.6 now. Thanks for your great job.
